### DISCUSSION TITLE
✅ mashREPL

### DISCUSSION THREAD

**Discussion by klue-less** [isontheline/pro.webssh.net#360] [Q&A] [answered]
Once I connect via SSH or log into a local computer, I have access to several different shells.

What is the goal of mashREPL in a small, lean, and mean piece of software such as webSSH?

I would avor removing mashREPL to reduce webSSH's attack surface and size of the body of code that must be maintained.

**Comment by klue-less** [comment]
Favor, not avor.

**Comment by mbierman** [comment, marked as answer]
I think the purpose is to have a minimal shell for doing things so that one does not first have to login to another machine to do some nominal thing. I don't see how this is an attack vector. Webssh is not there to protect your network. macOS, Windows, and Linux all have shells.  No one complains about that.

**Reply by isontheline** [reply]
I agree with you @mbierman that's a perfect resume of mashREPL : "the purpose is to have a minimal shell for doing things so that one does not first have to login to another machine to do some nominal thing"

* mashREPL is a very young addition and can't be used - yet - to do the same things that a "true" shell can do but it will be improved on new milestones.
* I also don't think that mashREPL could be an attack vector because it's not a shell, it's just a "fake" terminal because App Store rules doesn't allow a "true" terminal.
* mashREPL is just a sandboxed command line with some utilities
* mashREPL can't be launched from outside WebSSH
* mashREPL can't launch "real" commands that can do evil on you iOS system

@klue-less : Does a switch to disable mashREPL could be an option for you?

**Comment by isontheline** [comment]
Some people are already searching an exploit inside mashREPL.
Here is a deny of service which has been fixed in 14.17 : https://www.exploit-db.com/exploits/49883
